model/project_ref.go

```diff
@@ -2244,55 +2260,52 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
-// ValidateProjectCreation returns a boolean if you should surface the error or not.
-func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (bool, error) {
+// ValidateEnabledProjectsLimit takes in a merged project ref and validates if the current project is allowed to be enabled.
+// Returns a status code and error we are already at limit with enabled projects.
+func ValidateEnabledProjectsLimit(projectId string, config *evergreen.Settings, mergedRefToValidate ProjectRef) (int, error) {
 	if config.ProjectCreation.TotalProjectLimit == 0 || config.ProjectCreation.RepoProjectLimit == 0 {
-		return false, nil
+		return 0, nil
 	}
-	catcher := grip.NewBasicCatcher()
-	allEnabledProjects, err := GetEnabledProjects()
+	originalRef, err := FindMergedProjectRef(projectId, "", false)
 	if err != nil {
-		return true, errors.Wrap(err, "getting number of projects")
-	}
-	allEnabledProjectsId := []string{}
-	for _, p := range allEnabledProjects {
-		allEnabledProjectsId = append(allEnabledProjectsId, p.Id)
-	}
-	// No need to validate if project was enabled to begin with.
-	if utility.StringSliceContains(allEnabledProjectsId, projectId) {
-		return false, nil
+		return http.StatusInternalServerError, errors.Wrapf(err, "getting project '%s'", projectId)
 	}
 
-	// Only error if we are enabling a previously disabled project.
-	shouldError := false
-	pRef, err := findOneProjectRefQ(byId(projectId))
-	if err != nil {
-		return true, errors.Wrapf(err, "getting project '%s", projectId)
+	catcher := grip.NewBasicCatcher()
+	if originalRef == nil || !originalRef.IsEnabled() {
+		allEnabledProjects, err := GetNumberOfEnabledProjects()
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrap(err, "getting number of projects")
+		}
+		if originalRef == nil {
+			allEnabledProjects++
+		}
+		// Only validate total project limit if it's a new project or was previously disabled.
```

Comment:
Can you explain the "if it's a new project" part? Shouldn't we only validate if it's enabled period?

Reply:
This is also a confusing place for this comment since we don't check anything like that in this if statement.